File: smoothie-sample/src/main/java/com/example/smoothie/SimpleApp.java
package com.example.smoothie;

import android.app.Application;
import toothpick.Scope;
import toothpick.Toothpick;
import toothpick.configuration.Configuration;
import toothpick.smoothie.module.SmoothieApplicationModule;

public class SimpleApp extends Application {
  @Override
  public void onCreate() {
    super.onCreate();

    Toothpick.setConfiguration(Configuration.forProduction());

    Scope appScope = Toothpick.openScope(this);
    initToothpick(appScope);
  }

  public void initToothpick(Scope appScope) {
    appScope.installModules(new SmoothieApplicationModule(this));
  }
}

File: toothpick-kotlin-android-sample/src/main/kotlin/com/example/toothpick/kotlin/SimpleApp.kt
package com.example.toothpick.kotlin

import android.app.Application
import toothpick.Scope
import toothpick.Toothpick
import toothpick.configuration.Configuration
import toothpick.smoothie.module.SmoothieApplicationModule

class SimpleApp : Application() {
    override fun onCreate() {
        super.onCreate()

        Toothpick.setConfiguration(Configuration.forProduction())

        val appScope = Toothpick.openScope(this)
        initToothpick(appScope)
    }

    fun initToothpick(appScope: Scope) {
        appScope.installModules(SmoothieApplicationModule(this))
    }
}
